test/test.c

```diff
@@ -0,0 +1,128 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <stdbool.h>
+#include <sys/wait.h>
+
+#define COMMAND_LEN  512
+#define TESTCASE_LEN 32
+#define OUTPUT_LEN   1024
+
+typedef struct testcase_ {
+   char *name;
+   char *argv;
+   char *input;
+   char *output;
+} testcase;
+
+void raise_err(char *err_msg);
+
+int main(int argc, char **argv) {
+   if (argc != 2)
+      raise_err(
+         "test: usage:\n"
+         "\t./test\n"
+         "\t\tPATH_PROGRAM_SOURCE_CODE");
+   if (!system(NULL))
+      raise_err("test: no command processor available.");
+
+   /* Build a program to be tested. */
+   int statcode, retval;
+   char *command;
+   
+   command = malloc(COMMAND_LEN);
```

Comment:
We already know the size of `command`.
Why not `malloc((strlen("gcc -o object_program ") + strlen(argv[1])) * sizeof(char))`?